File: src/test/java/org/oewntk/yaml/in/TestsYamlModelLexGroups.java
/*
 * Copyright (c) 2021-2021. Bernard Bou.
 */

package org.oewntk.yaml.in;

import org.junit.BeforeClass;
import org.junit.Test;
import org.oewntk.model.LibTestModelLexGroups;

public class TestsYamlModelLexGroups
{
	@BeforeClass
	public static void init()
	{
		TestsYamlCommon.init();
	}

	@Test
	public void testCIMultipleAll()
	{
		LibTestModelLexGroups.testCIMultipleAll(TestsYamlCommon.model, TestsYamlCommon.ps);
	}

	@Test
	public void testCILemmas()
	{
		LibTestModelLexGroups.testCILemmas(TestsYamlCommon.model, "battle of verdun", TestsYamlCommon.ps);
	}

	@Test
	public void testCICounts()
	{
		LibTestModelLexGroups.testCICounts(TestsYamlCommon.model, "battle of verdun", TestsYamlCommon.ps);
	}

	@Test
	public void testCICountsFromMap()
	{
		LibTestModelLexGroups.testCICountsFromMap(TestsYamlCommon.model, "battle of verdun", TestsYamlCommon.ps);
	}

	@Test
	public void testCIHypermapWest()
	{
		LibTestModelLexGroups.testCIHypermap3(TestsYamlCommon.model, "West", TestsYamlCommon.ps);
	}

	@Test
	public void testCIHypermapBaroque()
	{
		LibTestModelLexGroups.testCIHypermap3(TestsYamlCommon.model, "Baroque", TestsYamlCommon.ps);
	}

	@Test
	public void testCIAi()
	{
		LibTestModelLexGroups.testCILexesFor(TestsYamlCommon.model, "ai", TestsYamlCommon.ps);
	}

	@Test
	public void testCIBaroque()
	{
		LibTestModelLexGroups.testCILexesFor(TestsYamlCommon.model, "baroque", TestsYamlCommon.ps);
	}

	@Test
	public void testCIWest3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsYamlCommon.model, "West", TestsYamlCommon.ps);
	}

	@Test
	public void testCIBaroque3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsYamlCommon.model, "Baroque", TestsYamlCommon.ps);
	}

	@Test
	public void testCIAi3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsYamlCommon.model, "Ai", TestsYamlCommon.ps);
	}

	@Test
	public void testCIAbsolute3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsYamlCommon.model, "Absolute", TestsYamlCommon.ps);
	}
}

File: src/test/java/org/oewntk/yaml/in/TestsYamlModelLexGroups.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.yaml.`in`

import org.junit.BeforeClass
import org.junit.Test
import org.oewntk.model.Key2
import org.oewntk.model.Lemma
import org.oewntk.model.Lex
import org.oewntk.model.Lex.Groups.groupByLCLemmaThenByKey2
import org.oewntk.model.LibTestModelLexGroups.testCICounts
import org.oewntk.model.LibTestModelLexGroups.testCICountsFromMap
import org.oewntk.model.LibTestModelLexGroups.testCIHypermap3
import org.oewntk.model.LibTestModelLexGroups.testCILemmas
import org.oewntk.model.LibTestModelLexGroups.testCILexesFor
import org.oewntk.model.LibTestModelLexGroups.testCILexesFor3
import org.oewntk.model.LibTestModelLexGroups.testCIMultipleAll
import org.oewntk.yaml.`in`.LibTestsYamlCommon.model
import org.oewntk.yaml.`in`.LibTestsYamlCommon.ps

class TestsYamlModelLexGroups {

    @Test
    fun testCIMultipleAll() {
        testCIMultipleAll(model, ps)
    }

    @Test
    fun testCILemmas() {
        testCILemmas(model, Lemma("battle of verdun"), ps)
    }

    @Test
    fun testCICounts() {
        testCICounts(model, Lemma("battle of verdun"), ps)
    }

    @Test
    fun testCICountsFromMap() {
        testCICountsFromMap(model, Lemma("battle of verdun"), ps)
    }

    @Test
    fun testCIAi() {
        testCILexesFor(model, Lemma("ai"), ps)
    }

    @Test
    fun testCIBaroque() {
        testCILexesFor(model, Lemma("baroque"), ps)
    }

    @Test
    fun testCIWest3() {
        testCILexesFor3(model, Lemma("West"), ps)
    }

    @Test
    fun testCIBaroque3() {
        testCILexesFor3(model, Lemma("Baroque"), ps)
    }

    @Test
    fun testCIAi3() {
        testCILexesFor3(model, Lemma("Ai"), ps)
    }

    @Test
    fun testCIAbsolute3() {
        testCILexesFor3(model, Lemma("Absolute"), ps)
    }

    private val lexLCHyperMap: Map<Lemma, Map<Key2, Collection<Lex>>> by lazy { model.lexes.asSequence().groupByLCLemmaThenByKey2() }

    @Test
    fun testCIHypermapWest() {
        testCIHypermap3(lexLCHyperMap, Lemma("West"), ps)
    }

    @Test
    fun testCIHypermapBaroque() {
        testCIHypermap3(lexLCHyperMap, Lemma("Baroque"), ps)
    }

    companion object {

        @JvmStatic
        @BeforeClass
        fun init() {
            model
        }
    }
}
